streams/early_zip_readable_streams.ts

```diff
@@ -88,21 +88,20 @@ export function earlyZipReadableStreams<T>(
   return new ReadableStream<T>({
     async start(controller) {
       try {
-        loop:
         while (true) {
           for (const reader of readers) {
             const { value, done } = await reader.read();
             if (!done) {
               controller.enqueue(value!);
             } else {
               await Promise.all(readers.map((reader) => reader.cancel()));
-              break loop;
+              controller.close();
+              break;
```

Comment:
Looks like this doesn't break from outer while-loop anymore?

Reply:
Reverted. My bad.